### PR TITLE
KIWI-1504: Implements person-info endpoint

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
### What changed

Implements person-info endpoint

### Why did it change

To allow the use of shared claims by the frontend

### Screenshots

image

### Issue tracking

- [KIWI-1504](https://govukverify.atlassian.net/browse/KIWI-1504)

### PII logging

- [x] Verified that no PII data is being logged

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Add SSM params to other envs

[KIWI-1504]: https://govukverify.atlassian.net/browse/KIWI-1504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/PersonInfoRequestProcessor.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No session found with the session id: ${sessionId}`);
+  	}
+
+  	this.logger.appendKeys({
+  		govuk_signin_journey_id: session?.clientSessionId,
+  	});
+
+  	const name = getFullName(person.name);
+		const encryptedResponseValue = this.encryptResponse({ name });
```
Comment (ViveakV) [inline src/services/PersonInfoRequestProcessor.ts:66]: I know we lint before commit now so howcomes the formatting looks so funky here

**Comment by ViveakV** [comment]
Small suggestion to maybe update the response body otherwise LGTM 🥳

**Review by ViveakV — DISMISSED** [review]

File: src/services/PersonInfoRequestProcessor.ts (hunk 1 above)
Comment (Caitlin-cooling) [reply src/services/PersonInfoRequestProcessor.ts]: It's difference between spaces and tabs which the linter doesn't pick up 

image

File: src/tests/utils/ApiConstants.ts [hunk 2]
```diff
@@ -6,4 +6,5 @@ export const constants = {
 	DEV_BAV_TEST_HARNESS_URL: process.env.DEV_BAV_TEST_HARNESS_URL,
 	DEV_BAV_SESSION_TABLE_NAME: "session-bav-cri-ddb",
 	DEV_BAV_PERSONAL_IDENTITY_TABLE_NAME: "person-identity-bav-cri-ddb",
+	PERSON_INFO_PRIVATE_KEY: process.env.PERSON_INFO_PRIVATE_KEY,
```
Comment (Caitlin-cooling) [inline src/tests/utils/ApiConstants.ts:9]: Follow up about putting private key in run-tests

Comment (Caitlin-cooling) [reply src/tests/utils/ApiConstants.ts:9]: Add SSM param to all envs

File: deploy/template.yaml [hunk 3]
```diff
@@ -848,6 +848,186 @@ Resources:
...
+            KeyId:
+              Fn::ImportValue: !Sub "${L2DynamoStackName}-session-table-key-id"
+        - Statement:
+            - Sid: KMSPolicy
+              Effect: Allow
+              Action:
+                - kms:*
+              Resource:
+                "*"
+        - Statement:
```
Comment (ViveakV) [inline deploy/template.yaml:892]: Why do we need TxMA stuff in here?

File: src/PersonInfoHandler.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+
+	async handler(event: APIGatewayProxyEvent, context: any): Promise<Response> {
+		logger.setPersistentLogAttributes({});
+		logger.addContext(context);
+
+		try {
+			const sessionId = this.validateEvent(event);
+			const publicKeyPath = checkEnvironmentVariable(EnvironmentVariables.PUBLIC_KEY_SSM_PATH, logger);
+    	PUBLIC_KEY = await getParameter(publicKeyPath);
```
Comment (ViveakV) [inline src/PersonInfoHandler.ts:37]: If you add a check for PUBLIC_KEY at this point you won't always need it fetch it cause it'll be in Lambda memory right?

File: src/services/PersonInfoRequestProcessor.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  	const session = await this.BavService.getSessionById(sessionId);
+
+		if (!person) {
+  		this.logger.error("No person found for session id", {
+  			messageCode: MessageCodes.PERSON_NOT_FOUND,
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No person found with the session id: ${sessionId}`);
+  	}
+
+  	if (!session) {
```
Comment (ViveakV) [inline src/services/PersonInfoRequestProcessor.ts:54]: [nit] do you think it's worth putting to session check 1st maybe? Just so if session exists but person doesn't we have govuk_signin_journey_id in logs to lookup, having person 1st doesn't give us any obvious benefits

Comment (ViveakV) [reply src/services/PersonInfoRequestProcessor.ts:54]: Shame we're having to query session DB just got the govuk_signin_journey_id though :/

File: deploy/template.yaml (hunk 3 above)
Comment (Caitlin-cooling) [reply deploy/template.yaml]: We don't, removed

File: src/PersonInfoHandler.ts (hunk 4 above)
Comment (Caitlin-cooling) [reply src/PersonInfoHandler.ts]: Yeah good point, added

File: src/services/PersonInfoRequestProcessor.ts (hunk 5 above)
Comment (Caitlin-cooling) [reply src/services/PersonInfoRequestProcessor.ts]: Swapped!

**Review by ViveakV — APPROVED** [review]

File: src/PersonInfoHandler.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+
+	async handler(event: APIGatewayProxyEvent, context: any): Promise<Response> {
+		logger.setPersistentLogAttributes({});
+		logger.addContext(context);
+
+		try {
+			const sessionId = this.validateEvent(event);
+			const publicKeyPath = checkEnvironmentVariable(EnvironmentVariables.PUBLIC_KEY_SSM_PATH, logger);
+			PUBLIC_KEY = PUBLIC_KEY ?? await getParameter(publicKeyPath);
```
Comment (ViveakV) [inline src/PersonInfoHandler.ts:37]: 🥳

**Review by ViveakV — DISMISSED** [review]

**Review by ViveakV — APPROVED** [review]

**Pull request merged**